modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java

```diff
@@ -0,0 +1,205 @@
+ * details.
+ */
+
+package com.liferay.media.object.apio.internal.architect.resource.test;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.liferay.apio.architect.file.BinaryFile;
+import com.liferay.apio.architect.pagination.PageItems;
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.media.object.apio.architect.model.MediaObject;
+import com.liferay.portal.apio.test.util.PaginationRequest;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerTestRule;
+
+import java.io.ByteArrayInputStream;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Ruben Pulido
+ */
+@RunWith(Arquillian.class)
+public class DefaultMediaObjectNestedCollectionResourceTest
+	extends BaseMediaObjectNestedCollectionResourceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testAddMediaObject() throws Exception {
+		FileEntry fileEntry = addMediaObject(
+			_group.getGroupId(),
+			new MediaObjectImpl(
+				new BinaryFile(
+					new ByteArrayInputStream(_CONTENT.getBytes(UTF_8)), 0L,
+					"application/octet-stream"),
+				"My media object testAddMediaObject",
+				"My media object description", null, null));
+
+		Assert.assertNotNull(fileEntry.getCreateDate());
+		Assert.assertEquals(
+			"My media object description", fileEntry.getDescription());
+		Assert.assertNotNull(fileEntry.getExtension());
+		Assert.assertNotNull(fileEntry.getMimeType());
+		Assert.assertNotNull(fileEntry.getModifiedDate());
+		Assert.assertEquals(fileEntry.getSize(), 0L);
+		Assert.assertEquals(
+			"My media object testAddMediaObject", fileEntry.getTitle());
+	}
+
+	@Test
+	public void testGetFileEntryPreviewURL() throws Exception {
+		FileEntry fileEntry = addMediaObject(
+			_group.getGroupId(),
+			new MediaObjectImpl(
+				new BinaryFile(
+					new ByteArrayInputStream(_CONTENT.getBytes(UTF_8)), 0L,
+					"application/octet-stream"),
+				"My media object testGetFileEntryPreviewURL", null, null,
+				null));
+
+		String fileEntryPreview = getFileEntryPreviewURL(fileEntry);
+
+		Assert.assertNotNull(fileEntryPreview);
```

Comment:
Just notNull? Couldn't we test anything about the entryPreview?

Reply:
After discussion, we'll test the content returned by contentUrl in client tests